File: main/src/main/java/com/bun133/tpprotector/TpProtector.java
package com.bun133.tpprotector;

import com.bun133.tpprotector.commands.TPCommand;
import org.bukkit.plugin.java.JavaPlugin;

public final class TpProtector extends JavaPlugin {
    Protector protector;
    TPCommand command;
    @Override
    public void onEnable() {
        // Plugin startup logic
        this.protector = new Protector();
        this.getServer().getPluginManager().registerEvents(protector,this);
        this.command = new TPCommand(protector);
        this.getCommand("tp").setExecutor(command);
        this.getServer().getScheduler().runTaskTimer(this,protector,1,1);
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }
}

File: main/src/main/kotlin/com/bun133/tpprotector/TpProtector.kt
package com.bun133.tpprotector

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.plugin.java.JavaPlugin

class TpProtector : JavaPlugin(), Listener {
    override fun onEnable() {
        // Plugin startup logic
        server.pluginManager.registerEvents(this, this)
    }

    @Override
    @EventHandler
    fun onDamage(e: EntityDamageEvent) {
        if (e.cause === EntityDamageEvent.DamageCause.CRAMMING)
            if (e.entity is Player)
                e.isCancelled = true
    }
}